addons/web/static/src/core/domain_tree.js

```diff
@@ -335,3 +362,231 @@ export function toTree(domain, options = {}) {
+                    };
+                }
+            }
+            case 4 /* List */: {
+                const tree = ast.value.length === 1 && _construcTree(ast.value[0], true);
+                if (tree?.type !== "expression") {
+                    return tree;
+                }
+                const expr = new Expression(ast);
+                if (!forceTree) {
+                    return expr;
+                }
+                return {
+                    type: "expression",
+                    value: expr.toString(),
+                };
+            }
+            case 8 /* FunctionCall */: {
+                if (ast.fn.value === "any") {
+                    const tree = _construcTree(ast.args[0]);
+                    if (tree.type !== "expression") {
+                        return tree;
+                    }
+                }
+                const expr = new Expression(ast);
+                if (!forceTree) {
+                    return expr;
+                }
+                return {
+                    type: "expression",
+                    value: expr.toString(),
+                };
+            }
+            case 6 /* UnaryOperator */: {
+                const tree = _construcTree(ast.right, true);
+                if (!tree.negate && typeof tree.operator === "string" && TERM_OPERATORS_NEGATION[tree.operator]) {
+                    tree.operator = TERM_OPERATORS_NEGATION[tree.operator];
+                } else {
+                    tree.negate = !tree.negate;
+                }
+                return tree;
+            }
+            case 7 /* BinaryOperator */: {
+                const left = _construcTree(ast.left);
+                const right = _construcTree(ast.right);
+                if (!left?.isPath() && !right?.isPath()) {
+                    return new Expression(ast);
+                }
+                switch (ast.op) {
+                    case "<":
+                    case "<=":
+                    case ">":
+                    case ">=":
+                    case "in":
+                    case "not in":
+                    case "!=":
+                    case "is":
+                    case "is not":
+                    case "==": {
+                        let operator = ast.op;
+                        if (operator === "==" || operator === "is") {
+                            operator = "=";
+                        }
+                        if (operator === "is not") {
+                            operator = "!=";
+                        }
+                        return {
+                            type: "condition",
+                            path: left.toString(),
+                            operator: operator,
+                            value: right,
+                        };
+                    }
+                    default:
+                        return new Expression(ast);
+                }
+            }
+            case 13 /* BinaryOperator */: {
```

Comment:
This is handling the if expression right?
```suggestion
            case 13 /* IfExpr */: {
```